File: sorting/shellSort.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
void shellSort(int *arr,int size)
{
int y,num,z,diff;
diff=size/2;
while(diff>0)
{
y=0+diff;
while(y<size)
{
num=arr[y];
z=y-diff;
while(z>=0 && arr[z]>num)
{
arr[z+diff]=arr[z];
z=z-diff;
}
arr[z+diff]=num;
y+=diff;
}
diff=diff/2;
}
}
void show(int *arr,int size)
{
for(int i=0;i<size;i++) printf("%d\n",arr[i]);
}
int main()
{
int req;
int *arr;
printf("Enter Requirement - ");
scanf("%d",&req);
if(req<=0)
{
printf("Invalid Requirement");
return 0;
}
arr=(int *)malloc(sizeof(int)*req);
if(arr==NULL)
{
printf("Unable To Allocate Memory");
return 0;
}
for(int i=0;i<req;i++)
{
printf("Enter A Number - ");
scanf("%d",&arr[i]);
}
shellSort(arr,req);
printf("----------After Sorting----------\n");
show(arr,req);
free(arr);
return 0;
}

File: JAVA_CODE/sorting/ShellSort.java
class ShellSort
{
public static void sort(int arr[])
{
int z,y,num,diff;
diff=arr.length/2;
while(diff>=1)
{
y=diff;
while(y<=arr.length-1)
{
num=arr[y];
z=y-diff;
while(z>=0 && arr[z]>num)
{
arr[z+diff]=arr[z];
z-=diff;
}
arr[z+diff]=num;
y++;
}
diff/=2;
}
}
public static void main(String args[])
{
int arr[]={45,12,78,65,98,32,55,60,81,99};
sort(arr);
for(int i=0;i<arr.length;i++) System.out.print(arr[i]+" ");
}
}